metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -690,36 +727,18 @@ public void run() throws Exception {
             }
 
             // Poll again after some time
-            long deadline = time.nanoseconds() + NANOSECONDS.convert(1, SECONDS);
+            long deadline = time.nanoseconds() + NANOSECONDS.convert(pollTimeSupplier.nextPollTimeMs(), MILLISECONDS);
             eventQueue.scheduleDeferred(
                     "poll",
                     new EventQueue.DeadlineFunction(deadline),
                     new PollEvent());
         }
     }
 
-    static String recordBatchToString(Collection<ApiMessageAndVersion> batch) {
-        String batchString = batch.stream().map(apiMessageAndVersion -> {
-            if (apiMessageAndVersion.message().apiKey() == MetadataRecordType.CONFIG_RECORD.id()) {
-                StringBuilder sb = new StringBuilder();
-                sb.append("ApiMessageAndVersion(");
-                ConfigRecord record = (ConfigRecord) apiMessageAndVersion.message();
-                sb.append("ConfigRecord(");
-                sb.append("resourceType=");
-                sb.append(record.resourceType());
-                sb.append(", resourceName=");
-                sb.append(record.resourceName());
-                sb.append(", name=");
-                sb.append(record.name());
-                sb.append(")");
-                sb.append(" at version ");
-                sb.append(apiMessageAndVersion.version());
-                sb.append(")");
-                return sb.toString();
-            } else {
-                return apiMessageAndVersion.toString();
-            }
-        }).collect(Collectors.joining(","));
+    String batchToRedactedString(Collection<ApiMessageAndVersion> batch) {
```

Comment:
Hmm, I think one record per log line would be better. Too hard to follow otherwise. And we might end up creating some super-giant strings that would break stuff.